### PR TITLE
fix - unhandled render format did not reset crop values

### PR DESCRIPTION
**Pull request by mrhipp**
fix - unhandled render format (e.g. RENDER_FMT_DXVA) did not reset crop  values

### CODE REVIEW

**Comment by Voyager1** [comment]
as far as I can tell, this doesn't really do much except reset the default values and protect against calling ```CDVDPlayerVideo::AutoCrop(DVDVideoPicture *pPicture, RECT &crop)``` for an unsupported render format. Correct?

File: xbmc/cores/dvdplayer/DVDPlayerVideo.cpp [hunk 1]
```diff
@@ -1191,16 +1191,9 @@ int CDVDPlayerVideo::OutputPicture(const DVDVideoPicture* src, double pts)
 
 void CDVDPlayerVideo::AutoCrop(DVDVideoPicture *pPicture)
 {
-  if ((pPicture->format == RENDER_FMT_YUV420P) ||
-     (pPicture->format == RENDER_FMT_NV12) ||
-     (pPicture->format == RENDER_FMT_YUYV422) ||
-     (pPicture->format == RENDER_FMT_UYVY422))
-  {
```
Comment (FernetMenta) [inline xbmc/cores/dvdplayer/DVDPlayerVideo.cpp]: why don't you reset to default in the else path of this block?

Comment (mrhipp) [inline xbmc/cores/dvdplayer/DVDPlayerVideo.cpp]: If find that code style is harder to read.

**Comment by mrhipp** [comment]
Voyager1: I some how manged to get crop settings for a  RENDER_FMT_DXVA video that i was unable remove unless i did a manual delete in the database after playing a supported format video with  
autocrop on.

**Comment by Voyager1** [comment]
fair enough, but if the Autocrop function with two parameters isn't called anywhere else I would recommend you fix as @FernetMenta suggested, for better readability. I personally also don't like the use of && as a surrogate to "if" (bad readability).

**Comment by mrhipp** [comment]
should now be according to best practice
http://elegantcode.com/2009/08/14/observations-on-the-if-statement/

**Comment by FernetMenta** [comment]
a much better place to reset the parameters is at the beginning of OutputPicture where we already check if the format has changed. If the format has changed we re-configure renderer and can also do things like resetting cropping.

**Comment by mrhipp** [comment]
Sorry, but I am not in for a complete rewrite beacuse of this bug. You have to take as it is or not.

**Comment by FernetMenta** [comment]
superseeded by #6154

**Comment by Voyager1** [comment]
come on, what are you guys thinking... I still think @mrhipp 's fix was ok at least for what he's trying to accomplish. Kill features because you don't like them?

**Comment by FernetMenta** [comment]
not because don't like them but because of nightmare implementation practices.

**Pull request closed without merging**